### ISSUE TITLE
After masterd received SIGTERM, job processor should stop jobs waiting for locks

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#352]
Originally reported of Google Code with ID 296.

```
Commit 2d6b541 added a clean daemon shutdown, used by masterd. As a consequence the job queue's worker pool does not start new jobs once SIGINT has been received, but it will continue to process running ones. Unfortunately jobs waiting for locks are also considered as “running” by the workerpool and therefore eventually start running. There should be a check similar to how there is for cancelling.
```
Originally added on 2012-10-31 17:22:24 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
It's SIGTERM, not SIGINT.
```
Originally added on 2012-10-31 17:44:59 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 942e226
Author: Michael Hanselmann <hansmi@google.com>
Date:   Thu Nov 1 14:54:03 2012 +0100

    jqueue: Return jobs to queue when shutting down
    
    When a job is still waiting for locks and the queue is shutting down,
    they should be returned and not actually start processing. Until now
    jobs which transitioned from “queued” to “waiting” were already
    considered to be running as far as the shutdown code was concerned.
    
    This fixes issue 296.
    
    Signed-off-by: Michael Hanselmann <hansmi@google.com>
    Reviewed-by: Iustin Pop <iustin@google.com>

```
Originally added on 2012-11-01 17:46:35 +0000 UTC.

Mentioned Issue with GoogleCode ID 296 has been migrated to GitHub with ID #352.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking as released the bugs fixed in 2.6.2.
```
Originally added on 2013-01-11 12:13:05 +0000 UTC.

Changed State: Released
